wpimath/src/test/native/cpp/geometry/Pose3dTest.cpp

```diff
@@ -221,3 +221,111 @@ TEST(Pose3dTest, TwistNaN) {
     EXPECT_FALSE(std::isnan(twist.rz.value()));
   }
 }
+
+TEST(Pose3dTest, Nearest) {
+  const Pose3d origin{0_m, 0_m, 0_m, Rotation3d{}};
+
+  // Distance sort
+  // each poseX is X units away from the origin at various positions in 3D
+  // space.
```

Comment:
Same comment as on Java- `pose4` and `pose5` aren't 4 and 5 units away from the origin.